src/game/tutorial.ts

```diff
@@ -514,7 +514,10 @@ export function getBoardInfo(
       return { ...data, boardState };
     }
 
-    const fen = boardToFEN(data.boardState);
+    if (data.boardState) {
+      const fen = boardToFEN(data.boardState);
+    }
+    
```

Comment:
This is probably fine to merge for now, because it at least fixes the existing bug.

But we should consider the case where (now that both `fen` and `boardState` are optional), someone accidentally defines a tutorial with neither. What happens in that case? It seems like something somewhere will break in some unexpected way.

Probably better would be to do a check: if `!data.fen && !data.boardState` then `throw new Error("either 'fen' or 'boardState' must be defined")`.. that way we can catch the problem earlier